src/lib/containers/HasAccess.tsx

```diff
@@ -55,65 +94,109 @@ export default class HasAccess extends React.Component<
+      default:
+        throw Error('downloadTypeEnum passed incorrectly')
+    }
+  }
+
+  // Utility to declare what type of download this is
+  getDownloadType = () => {
+    const { fileHandle } = this.props
     const { restrictionInformation } = this.state
-    const { synapseId } = this.props
-    const icon =
-      restrictionInformation &&
-      restrictionInformation.hasUnmetAccessRequirement ? (
-        <span className="fa-layers fa-fw" style={{ marginRight: 5 }}>
-          <FontAwesomeIcon
-            icon={faCircle}
-            className="SRC-warning-color"
-            size="lg"
-          />
-          <FontAwesomeIcon
-            icon={faMinusCircle}
-            className="SRC-half-opacity"
-            style={{ transform: 'translate(4%, -4%)' }}
-            size="xs"
-          />
-        </span>
-      ) : (
-        <span className="fa-layers fa-fw" style={{ marginRight: 5 }}>
-          <FontAwesomeIcon
-            icon={faCircle}
-            className="SRC-success-color"
-            size="lg"
-          />
-          <FontAwesomeIcon
-            icon={faUnlockAlt}
-            className="SRC-half-opacity"
-            style={{ transform: 'translate(4%, -4%)' }}
-            size="xs"
-          />
-        </span>
-      )
+    const isOpen =
+      fileHandle ||
+      (restrictionInformation &&
+        RestrictionLevel.OPEN === restrictionInformation.restrictionLevel)
+    const isTooLarge = fileHandle && fileHandle.contentSize >= GIGABYTE_SIZE
+    if (isTooLarge) {
+      return DownloadTypeEnum.TooLargeFile
+    } else if (isOpen) {
+      const concreteType = fileHandle ? fileHandle.concreteType : ''
+      const isGoogleCloudFileHandle =
+        concreteType === GoogleCloudFileHandleEnum.GoogleCloudFileHandle
+      if (isGoogleCloudFileHandle) {
+        return DownloadTypeEnum.CloudFileHandle
+      } else {
+        // @ts-ignore
+        const isExternalFileHandle = Object.values(
+          ExternalFileHandleConcreteTypeEnum,
+        ).includes(concreteType)
+        if (isExternalFileHandle) {
+          return DownloadTypeEnum.ExternalFileHandle
+        }
+        return DownloadTypeEnum.IsOpenNoRestrictions
+      }
+    } else {
+      return DownloadTypeEnum.NoAccess
+    }
+  }
 
+  // Show Access Requirements
+  renderARsLink = (downloadType: DownloadTypeEnum) => {
+    const { restrictionInformation } = this.state
+    const { synapseId, deniedAccess } = this.props
+    const hasUnmetAccessRequirement =
+      (restrictionInformation &&
+        restrictionInformation!.hasUnmetAccessRequirement) ||
+      ''
+    const restrictionLevel =
+      (restrictionInformation && restrictionInformation!.restrictionLevel) || ''
     let viewARsLink: React.ReactElement = <></>
     if (
-      restrictionInformation &&
-      RestrictionLevel.OPEN !== restrictionInformation.restrictionLevel
+      RestrictionLevel.OPEN !== restrictionLevel &&
+      downloadType !== DownloadTypeEnum.TooLargeFile
```

Comment:
Alternatively, you could still show the "Get Access"/"View Terms" link when file is too large.